hubspot-style/src/test/java/com/hubspot/immutables/utils/WireSafeEnumTest.java

```diff
@@ -76,6 +76,30 @@ public static CollidingJsonEnumWithCreator fromString(String s) {
     }
   }
 
+  public enum SubclassedEnum {
+    ABC,
+    DEF {
+      @Override
+      public String getSomething() {
+        return "b";
+      }
+    }
+    ;
+
+    public String getSomething() {
+      return "a";
+    }
+  }
```

Comment:
Clarity nit: this isn't a case of enum subclassing (which isn't permitted in java), it's an enum with method overrides.

Reply:
Hmm, i'm not sure that I understand what the difference is here?  `SubclassedEnum.DEF.getClass()` won't return `SubclassEnum`, it will return the subclass `SubclassEnum$1`, while `SubclassedEnum.ABC.getClass()` will return `SubclassEnum`

Reply:
I updated the names to reference the method overriding rather than the subclassing, is that better?

Reply:
👍 works for me. I don't think there's a huge difference:

> The optional class body of an enum constant implicitly defines an anonymous class declaration (§15.9.5) that extends the immediately enclosing enum type.